Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm

```diff
@@ -478,12 +466,11 @@
     if (NSString *cachedString = objectForKey<NSString>(m_resources, path))
         return cachedString;
 
-    bool isServiceWorker = backgroundContentIsServiceWorker();
-    if (!isServiceWorker && [path isEqualToString:generatedBackgroundPageFilename])
-        return generatedBackgroundContent();
+    if ([path isEqualToString:generatedBackgroundPageFilename] && !backgroundContentIsServiceWorker())
+        return [static_cast<NSString *>(generatedBackgroundContent()) dataUsingEncoding:NSUTF8StringEncoding];
 
-    if (isServiceWorker && [path isEqualToString:generatedBackgroundServiceWorkerFilename])
-        return generatedBackgroundContent();
+    if ([path isEqualToString:generatedBackgroundServiceWorkerFilename] && !backgroundContentIsServiceWorker())
+        return [static_cast<NSString *>(generatedBackgroundContent()) dataUsingEncoding:NSUTF8StringEncoding];
```

Comment:
Ditto.
```suggestion
        return generatedBackgroundContent();
```

Reply:
Actually these two ifs, this can be simplified.

```suggestion
if ([path isEqualToString: generatedBackgroundPageFilename] || [path isEqualToString:generatedBackgroundServiceWorkerFilename])
    return generatedBackgroundContent();
```